### ISSUE TITLE
telegraf doesn't resume sending data after long influxdb unavailability

### ISSUE THREAD

**Issue by r0bj** [influxdata/telegraf#795]
Telegraf doesn't resume sending data after long influxdb unavailability. Daemon is running but it doesn't generate any network activity.

Fragment of strace activity:
https://gist.github.com/r0bj/b00ab48a91f42ae3255d

After telegraf restart it works ok.

/etc/telegraf/telegraf.conf:
```
[[outputs.influxdb]]
  urls = ["http://influxdbhost:8086"]
  database = "telegraf" # required
  precision = "s"
```

**Comment by sparrc** [comment]
could you provide any logs?

**Comment by r0bj** [comment]
https://gist.github.com/r0bj/cdeef5975ec27c9dd4d3

Relevant information is that influxdb domain "prod.os-metrics-db.service.sjc.consul" was deregistered during influxdb unavailability so it was unresolvable at that time. Unfortunately after influxdb was up again and its domain was registered, telegraf didn't resume sending data.

**Comment by sparrc** [comment]
What version of telegraf?

**Comment by r0bj** [comment]
Telegraf - Version 0.10.2

**Comment by sparrc** [comment]
I noticed this behavior in that version but never quite pinned it down, but this was fixed in 0.10.3 when I overhauled the internal metric buffer handling, please upgrade
